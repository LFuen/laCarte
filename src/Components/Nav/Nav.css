
.navigation {
    display: flex;
    justify-content: center; 
}

/* .navigation > a{
    text-decoration: none;
    color: rgb(0, 0, 0);
    z-index: 1;
}

.navLink:before{
    content: "";
    position: absolute;
    width: 100%;
    height: 3px;
    left: 0;
    right: 0;
    background: #eb7474;
    visibility: hidden;
    border-radius: 8px;
    transform: scaleX(0);
    transition: .25s linear;
}

.navLink:hover:before{
    visibility: visible;
    transform: scaleX(1);
} */

