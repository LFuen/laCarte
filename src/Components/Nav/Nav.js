import React, {Component} from 'react'
import {Route, NavLink} from 'react-router-dom'
import Chefs from '../Chefs/Chefs'
import About from '../About/About'
import MealList from '../MealList/MealList'
import MealChoose from '../MealChoose/MealChoose'
import OrderForm from '../OrderForm/OrderForm'
import Cuisine from '../Cuisine/Cuisine'
import Ingredients from '../Ingredients/Ingredients'
import OrderPlaced from '../OrderPlaced/OrderPlaced'
import Welcome from '../Welcome/Welcome'
import ChefBio from '../ChefBio/ChefBio'
import LaCarteContext from '../../context/LaCarteContext'
import OrderHistory from '../OrderHistory/OrderHistory'
import OrderEdit from '../OrderEdit/OrderEdit'
import OrderUpdated from '../OrderUpdated/OrderUpdated'
import './Nav.css'


class Nav extends Component {

    static contextType = LaCarteContext;

    render() {

        return(
            <div>
                <nav className='navigation' id='navbar'>
                    <NavLink to='/' className='navLink'>Home</NavLink>
                    <span title='La Carte'><img className='chefHat' alt='chef' src='https://lfuen.github.io/laCarte/images/chef.png'/></span>
                    <NavLink to='/MealChoose' className='navLink'>Selection</NavLink>
                    <span title='La Carte'><img className='chefHat' alt='chef' src='https://lfuen.github.io/laCarte/images/chef.png'/></span>
                    <NavLink to='/Chefs' className='navLink'>Chefs</NavLink>
                    <span title='La Carte'><img className='chefHat' alt='chef' src='https://lfuen.github.io/laCarte/images/chef.png'/></span>
                    <NavLink to='/OrderHistory' className='navLink'>Orders</NavLink>
                    <span title='La Carte'><img className='chefHat' alt='chef' src='https://lfuen.github.io/laCarte/images/chef.png'/></span>
                    <NavLink to='/About' className='navLink'>About</NavLink>
                </nav>
                
                <Route exact path='/' component={Welcome}/>
                <Route exact path='/MealList' component={MealList}/>
                <Route exact path='/Chefs' component={Chefs}/>
                <Route exact path='/chefs/:chef' component={ChefBio}/>
                <Route exact path='/About' component={About}/>
                <Route exact path='/MealChoose' component={MealChoose}/>
                <Route exact path='/orders/:id' component={OrderForm}/>
                <Route exact path='/cuisines/:id/:origin' component={Cuisine}/>
                <Route exact path='/meals/:id' component={Ingredients}/>
                <Route exact path='/OrderPlaced' component={OrderPlaced}/>
                <Route exact path='/OrderHistory' component={OrderHistory}/>
                <Route exact path='/orders/:id/:meal' component={OrderEdit}/>
                <Route exact path='/OrderUpdated' component={OrderUpdated}/>
            </div>
        )
    }
}


export default Nav
