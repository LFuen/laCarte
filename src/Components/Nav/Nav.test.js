import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom';
import LaCarteContext from '../../context/LaCarteContext'
import Nav from './Nav'



describe(`Nav Component`, () => {
    describe(`Snapshot`, () => {
        it('renders the UI as expected', () => {
            const nav = renderer.create(<BrowserRouter><Nav /></BrowserRouter>).toJSON()
                expect(nav).toMatchSnapshot()
        })
    })

    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            
            const div = document.createElement('div');

            const value = {
                meals: 'string',
                chefs: 'string',
                cuisines: 'string',
                orders: 'string',
                addOrder: 'string',
                updateOrder: 'string',
                deleteOrder: 'string',
            };

            ReactDOM.render(
                <BrowserRouter>
                    <LaCarteContext.Provider value={value}>
                        <Nav />
                    </LaCarteContext.Provider>
                </BrowserRouter>,
                div)
            ReactDOM.unmountComponentAtNode(div)
        })
    })
})