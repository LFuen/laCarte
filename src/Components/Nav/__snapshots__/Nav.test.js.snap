// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component Snapshot renders the UI as expected 1`] = `
<div>
  <nav
    className="navigation"
    id="navbar"
  >
    <a
      aria-current="page"
      className="navLink active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Home
    </a>
    <span
      title="La Carte"
    >
      <img
        alt="chef"
        className="chefHat"
        src="https://lfuen.github.io/laCarte/images/chef.png"
      />
    </span>
    <a
      aria-current={null}
      className="navLink"
      href="/MealChoose"
      onClick={[Function]}
    >
      Selection
    </a>
    <span
      title="La Carte"
    >
      <img
        alt="chef"
        className="chefHat"
        src="https://lfuen.github.io/laCarte/images/chef.png"
      />
    </span>
    <a
      aria-current={null}
      className="navLink"
      href="/Chefs"
      onClick={[Function]}
    >
      Chefs
    </a>
    <span
      title="La Carte"
    >
      <img
        alt="chef"
        className="chefHat"
        src="https://lfuen.github.io/laCarte/images/chef.png"
      />
    </span>
    <a
      aria-current={null}
      className="navLink"
      href="/OrderHistory"
      onClick={[Function]}
    >
      Orders
    </a>
    <span
      title="La Carte"
    >
      <img
        alt="chef"
        className="chefHat"
        src="https://lfuen.github.io/laCarte/images/chef.png"
      />
    </span>
    <a
      aria-current={null}
      className="navLink"
      href="/About"
      onClick={[Function]}
    >
      About
    </a>
  </nav>
  <div
    className="welcomePage"
  >
    <span
      title="La Carte"
    >
      <img
        alt="LaCarte"
        className="titleImage"
        src="https://lfuen.github.io/laCarte/images/laCarteFont.png"
      />
    </span>
    <h3
      className="worked"
    >
      The Healthy-Meal-Alternative App for the Busy Professional.
    </h3>
    <h5>
      (We didn't say workaholic, you did.)
    </h5>
    <div
      className="flexed"
    >
      <span
        title="Bon Apetit!"
      >
        <img
          alt="laCarte"
          className="mainImage"
          src="https://lfuen.github.io/laCarte/images/laCarte.png"
        />
      </span>
    </div>
  </div>
</div>
`;
